Services/DataX.Config/DataX.Config/ConfigGeneration/Processor/S500_GenerateProjectionFile.cs

```diff
@@ -54,7 +56,8 @@ public override async Task<string> Process(FlowDeploymentSession flowToDeploy)
             var filePath = ResourcePathUtil.Combine(runtimeConfigBaseFolder, "projection.txt");
             var savedFile = await RuntimeStorage.SaveFile(filePath, finalProjections);
             var secretName = $"{config.Name}-projectionfile";
-            var savedSecretId = await KeyVaultClient.SaveSecretAsync(runtimeKeyVaultName, secretName, savedFile);
+            var uriPrefix = (Configuration[Constants.ConfigSettingName_SparkType].Length > 0 && Configuration[Constants.ConfigSettingName_SparkType] == Constants.SparkTypeDataBricks) ? Constants.PrefixSecretScope : Constants.PrefixKeyVault;
+            var savedSecretId = await KeyVaultClient.SaveSecretAsync(runtimeKeyVaultName, secretName, savedFile, uriPrefix);
```

Comment:
Rather than every caller computing the uriPrefix, it may be better to push that logic to KeyVaultClient and handle it there in one place.

Reply:
Done. Refactored it in KeyVaultClient